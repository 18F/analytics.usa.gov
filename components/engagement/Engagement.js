import React from "react";
import PropTypes from "prop-types";

import EngagementRate from "./EngagementRate";
import AverageEngagementDuration from "./AverageEngagementDuration";

function Engagement({ dataURL, dataPrefix }) {
  const dataHrefBase = `${dataURL}/${dataPrefix}`;

  return (
    <>
      <section id="average_engagement_time" className="desktop:grid-col-6">
        <div className="grid-row">
          <h4 className="grid-col-12">
            <a href="/definitions#report_historical_average_engagement_time">
              Average Engagement Time Per Session
            </a>
          </h4>
        </div>
        <div className="grid-row">
          <div className="grid-col-12">
            <AverageEngagementDuration dataHrefBase={dataHrefBase} />
          </div>
        </div>
      </section>

      <section id="engagement_rate" className="desktop:grid-col-6">
        <div className="grid-row">
          <h4 className="grid-col-12">
            <a href="/definitions#report_historical_engagement_rate">
              Percent of Engaged Sessions
            </a>
          </h4>
        </div>
        <div className="grid-row">
          <div className="data grid-col-12">
            <EngagementRate dataHrefBase={dataHrefBase} />
          </div>
        </div>
      </section>
    </>
  );
}

Engagement.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
};

export default Engagement;
