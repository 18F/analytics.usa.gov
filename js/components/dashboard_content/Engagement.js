import React from "react";
import PropTypes from "prop-types";

import EngagementRate from "./EngagementRate";
import AverageEngagementDuration from "./AverageEngagementDuration";

/**
 * Contains charts and other data visualizations for the user engagement section
 * of the site. This component is mainly laying out the structure for the
 * section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function Engagement({ dataHrefBase }) {
  return (
    <>
      <section id="average_engagement_time" className="desktop:grid-col-6">
        <div className="grid-row">
          <h4 className="grid-col-12">
            <a href="/definitions#report_historical_average_engagement_time">
              Average Engagement Time Per Session
            </a>
          </h4>
        </div>
        <div className="grid-row">
          <span className="grid-col-12">
            <AverageEngagementDuration dataHrefBase={dataHrefBase} />
          </span>
        </div>
      </section>

      <section id="engagement_rate" className="desktop:grid-col-6">
        <div className="grid-row">
          <h4 className="grid-col-12">
            <a href="/definitions#report_historical_engagement_rate">
              Percent of Engaged Sessions
            </a>
          </h4>
        </div>
        <div className="grid-row">
          <span className="data grid-col-12">
            <EngagementRate dataHrefBase={dataHrefBase} />
          </span>
        </div>
      </section>
    </>
  );
}

Engagement.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default Engagement;
