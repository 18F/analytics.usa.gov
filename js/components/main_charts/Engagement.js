import React from "react";
import PropTypes from "prop-types";

import EngagementRate from "./EngagementRate";
import AverageEngagementDuration from "./AverageEngagementDuration";

function Engagement({ dataHrefBase }) {
  return (
    <>
      <section id="average_engagement_time" className="desktop:grid-col-6">
        <div className="grid-row">
          <h4 className="grid-col-12">
            <a href="/definitions#report_historical_average_engagement_time">
              Average Engagement Time Per Session
            </a>
          </h4>
        </div>
        <div className="grid-row">
          <span className="grid-col-12">
            <AverageEngagementDuration dataHrefBase={dataHrefBase} />
          </span>
        </div>
      </section>

      <section id="engagement_rate" className="desktop:grid-col-6">
        <div className="grid-row">
          <h4 className="grid-col-12">
            <a href="/definitions#report_historical_engagement_rate">
              Percent of Engaged Sessions
            </a>
          </h4>
        </div>
        <div className="grid-row">
          <span className="data grid-col-12">
            <EngagementRate dataHrefBase={dataHrefBase} />
          </span>
        </div>
      </section>
    </>
  );
}

Engagement.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default Engagement;
