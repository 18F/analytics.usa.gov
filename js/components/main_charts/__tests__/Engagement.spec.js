import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import Engagement from "../Engagement";

describe("Engagement", () => {
  let component;

  beforeEach(async () => {
    component = await render(
      <Engagement dataHrefBase="http://www.example.com/data/" />,
    );
    await waitFor(() => screen.getByText("Percent of Engaged Sessions"));
  });

  it("renders", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
