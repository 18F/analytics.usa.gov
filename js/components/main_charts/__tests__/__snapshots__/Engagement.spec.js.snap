// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement renders 1`] = `
[
  <section
    className="desktop:grid-col-6"
    id="average_engagement_time"
  >
    <div
      className="grid-row"
    >
      <h4
        className="grid-col-12"
      >
        <a
          href="/definitions#report_historical_average_engagement_time"
        >
          Average Engagement Time Per Session
        </a>
      </h4>
    </div>
    <div
      className="grid-row"
    >
      <span
        className="grid-col-12"
      >
        <div>
          <div
            className="data"
          />
        </div>
      </span>
    </div>
  </section>,
  <section
    className="desktop:grid-col-6"
    id="engagement_rate"
  >
    <div
      className="grid-row"
    >
      <h4
        className="grid-col-12"
      >
        <a
          href="/definitions#report_historical_engagement_rate"
        >
          Percent of Engaged Sessions
        </a>
      </h4>
    </div>
    <div
      className="grid-row"
    >
      <span
        className="data grid-col-12"
      >
        <div>
          <div
            className="data"
          />
        </div>
      </span>
    </div>
  </section>,
]
`;
